### DISCUSSION TITLE
Feature ideas

### DISCUSSION THREAD

**Discussion by alexbarry** [alexbarry/AlexCalc#1] [Ideas]
Some things that I think would be easy enough to implement if there is interest:

* output formatting: number of decimal places, commas or other delimiters for thousands (or tens of thousands, which I think some places use). Also showing engineering notation (output in form of `x * 10^(3*n)` instead of `x * 10^n`) (I meant to include some output formatting before releasing it, but I wanted to try sharing it first to see if there was any interest)
* defining functions (similar to how variables are defined)
* graphing those functions
* simple equation solving (this might be a lot harder, but I could probably implement some simple methods)
* UI customizations (different colour themes, configurable buttons)
* horizontal layout
* unit output preferences (e.g. prefer imperial for distance, show astronomical units or parsecs if greater than 10^16 metres or something like that)

**Comment by Blocky-mint** [comment]
I think most of the feature sounds good. Even if i don't need most of them for now, but in the future i could see myself needing those. Perhaps it would be better to separate complex feature in their own section (i saw some calculator app that does this) but i think its probably gonna be difficult to implement

Here is some example of the calculator app that have separate mode:

https://play.google.com/store/apps/details?id=advanced.scientific.calculator.calc991.plus

https://play.google.com/store/apps/details?id=com.scientificCalculator

However, one suggestion for me would be to create an auto toggle for layout that can detect whether the device are in horizontal or vertical mode (it would probably be difficult) so a manual mode that user can choose whether they want horizontal or vertical layout.
